rules-reviewed/openshift/local-storage.windup.xml

```diff
@@ -141,10 +141,10 @@
             <where param="path1">
                 <!-- absolute paths on windows -->
                 <!-- UPDATE TO REMOVE PATTERNS STARTING WITH java: AND classpath: -->
-                <matches pattern="(([a-zA-Z]):)(?&lt;![\&lt;\\\/\d\w])([\\\/]\w+)+(\.\w+)?" />
+                <matches pattern="(([:=(])([ ])*([&quot;'])?([a-zA-Z]):)(?&lt;![\&lt;\\\/\d\w])([\\\/]\w+)+(\.\w+)?"/>
```

Comment:
Can you explain what this pattern does now? I'm having a hard time with the first bit of it.

Reply:
Sure, i tried to change the pattern to identify only code that really contains a path definition, so:
- `([:=(])` matches the typical way of declaring a variable (i.e. `:` and `=`) or the method invocation delimiter `(`
- `([ ])*` matches any optional blank space because it depends on the coding style the way of declaring variables (i.e. `String path="C:/tmp"` and `String path = "C:/tmp"` are both valid)
- `([&quot;'])?` matches `"` or `'` as they're the delimiter for string in Java and properties files
- then there's the same pattern we already had